awesome.c

```diff
@@ -901,20 +913,44 @@ main(int argc, char **argv)
 
     ewmh_init_lua();
 
+    /* Parse and run configuration file before adding the screens */
+    if (globalconf.no_auto_screen)
+    {
+        /* Disable automatic screen creation, awful.screen has a fallback */
+        globalconf.ignore_screens = true;
+
+        if(!luaA_parserc(&xdg, confpath))
+            fatal("couldn't find any rc file");
+    }
+
+    screen_emit_scanning();
```

Comment:
Because there is more PRs building on this. I can add more commits to this one, but it's already enormous ;). As you why not add the signal in those PRs: because users/modules can use it sonner.

The general idea is to have a better defined initialization for modules. With better defined steps, things like handling startup errors become a lot easier. In each steps, some "things", like having screen objects, is well defined. Given screens have 2 modes and different signals depending on the modes, it makes some sense to have both `scanning` and `scanned` for both client and screens.